features/correction/service.py

```
@@ -95,9 +100,35 @@ async def _run_rag(self, correction_id: str) -> None:
+                logger.exception(
+                    "RAG 인사이트 생성 실패 (correction_id: %s): %s", correction_id, exc
+                )
+            else:
+                logger.exception("RAG 처리 실패 (correction_id: %s): %s", correction_id, exc)
             await self._mark_failed(correction_id)
 
+    async def retry(self, correction_id: str, background_tasks: BackgroundTasks) -> None:
+        """실패한 첨삭을 마지막 단계 기준으로 재시도"""
+        correction = await self._repository.get_by_id(correction_id)
+        if correction is None:
+            raise ValueError(f"첨삭을 찾을 수 없습니다: {correction_id}")
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)